### PR TITLE
EM-6199  add audit task

### PR DESCRIPTION
**Pull request by mdayican**
### Jira link
[EM-6199](https://tools.hmcts.net/jira/browse/EM-6199) add new report to send monthly

### CODE REVIEW

**Review by MichaelJacob21012 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            .create()
+            .setHeader(HEARING_AUDIT_SUMMARY_CSV_HEADERS)
+            .build();
+        try (
+            FileWriter fileWriter = new FileWriter(csvFile);
+            CSVPrinter printer = new CSVPrinter(fileWriter, csvFileHeader)
+        ) {
+            for (AuditEntry entry : Optional.ofNullable(data).orElse(emptyList())) {
+                if (entry instanceof HearingRecordingSegmentAuditEntry hearingAudit) {
+                    hearingAudit = (HearingRecordingSegmentAuditEntry) entry;
```
Comment (yogesh-hullatti) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java:44]: We no more need this casting. Please remove this line.

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                        hearingRecSeg.getHearingRecording().getHearingSource(),
+                        hearingRecSeg.getHearingRecording().getServiceCode(),
+                        (int) Math.ceil((float) hearingRecSeg.getFileSizeMb() / 1000),
+                        hearingRecSeg.getHearingRecording().getCcdCaseId(),
+                        hearingRecSeg.getCreatedOn(),
+                        "",
+                        ""
+                    );
+                } else if (entry instanceof HearingRecordingAuditEntry hearingRecordingAudit) {
+                    hearingRecordingAudit = (HearingRecordingAuditEntry) entry;
```
Comment (yogesh-hullatti) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java:60]: We no more need this casting. Please remove this line.

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                        hearingRec.getHearingSource(),
+                        hearingRec.getServiceCode(),
+                        "",
+                        hearingRec.getCcdCaseId(),
+                        hearingRec.getCreatedOn(),
+                        "",
+                        ""
+                    );
+                } else if (entry instanceof HearingRecordingShareeAuditEntry hearingShareeAudit) {
+                    hearingShareeAudit = (HearingRecordingShareeAuditEntry) entry;
```
Comment (yogesh-hullatti) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java:76]: We no more need this casting. Please remove this line.

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java]: removed

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java (hunk 3 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java]: removed

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java]: removed

**Review by yogesh-hullatti — APPROVED** [review]

**Pull request merged**